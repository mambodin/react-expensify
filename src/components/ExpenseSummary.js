import React from 'react';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';


export const ExpenseSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense ': 'expenses' ;
    const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
};


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount : visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);