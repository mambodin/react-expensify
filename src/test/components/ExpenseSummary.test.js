import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseSummary}  from '../../components/ExpenseSummary';



test('should render ExpenseSummary with 1 Expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235} />);

    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with  multiple Expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal={23425235} />);

    expect(wrapper).toMatchSnapshot();
});