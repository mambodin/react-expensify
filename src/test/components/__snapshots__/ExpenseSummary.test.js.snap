// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with  multiple Expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $234,252.35
  </h1>
</div>
`;

exports[`should render ExpenseSummary with 1 Expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense 
     totalling 
    $2.35
  </h1>
</div>
`;
